caddyhttp/staticfiles/fileserver.go

```diff
@@ -121,13 +125,36 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 		return http.StatusNotFound, nil
 	}
 
+	filename := d.Name()
+
+encodings:
+	for _, encoding := range staticEncodingPriority {
+		if strings.Contains(r.Header.Get("Accept-Encoding"), encoding) {
+			gf, err := fs.Root.Open(location + staticEncoding[encoding])
+			if err == nil {
+				defer gf.Close()
+
+				stat, err := gf.Stat()
+				if err == nil {
+					d = stat
+					f = gf
+
+					w.Header().Add("Vary", "Accept-Encoding")
+					w.Header().Set("Content-Encoding", encoding)
+
+					break encodings
```

Comment:
Did you intend for this to be a labeled break? I guess I like that it's readable, but it's not a nested loop.